File: sixthSeminar/src/main/java/org/sopt/sixthSeminar/domain/SOPT.java
package org.sopt.sixthSeminar.domain;

import jakarta.persistence.Embeddable;
import jakarta.persistence.Enumerated;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.sopt.thirdSeminar.domain.enums.Part;

import static jakarta.persistence.EnumType.STRING;

@Embeddable
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class SOPT {
    private int generation;

    @Enumerated(value = STRING)
    private Part part;
}

File: secondKotlin/src/main/kotlin/org/sopt/secondKotlin/domain/SOPT.kt
package org.sopt.secondKotlin.domain

import jakarta.persistence.Embeddable
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import org.sopt.secondKotlin.domain.enums.Part

@Embeddable
class SOPT(
    private val generation : Int,

    @Enumerated(value = EnumType.STRING)
    private val part : Part
) {
}